pkg/runtime/pubsub/cloudevents.go

```diff
@@ -14,28 +14,34 @@ limitations under the License.
 package pubsub
 
 import (
-	"github.com/google/uuid"
-
 	contribContenttype "github.com/dapr/components-contrib/contenttype"
 	contribPubsub "github.com/dapr/components-contrib/pubsub"
 )
 
 // CloudEvent is a request object to create a Dapr compliant cloudevent.
+// The cloud event properties can manually be overwritten by using metadata beginning with "cloudevent-" as prefix.
 type CloudEvent struct {
-	ID              string
-	Data            []byte
-	Topic           string
-	Pubsub          string
-	DataContentType string
-	TraceID         string
-	TraceState      string
+	ID              string `mapstructure:"cloudevent-id"`
+	Data            []byte `mapstructure:"cloudevent-data"`
+	Topic           string `mapstructure:"cloudevent-topic"`
```

Comment:
Not sure if we want people to override properties like data, topic, etc? Sounds dangerous as people could shoot themselves in the foot real bad (and then we'd have to support a variety of edge cases)

Reply:
Overriding data is weird, but it would have to be done explicitly. There is no reason to prevent it.

Topic is fine for overriding - this topic here isn't the topic used for routing actually.